Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -1932,13 +1932,16 @@ public boolean moveToZone(Zone toZone, Ability source, Game game, boolean flag,
 
     @Override
     public boolean moveToExile(UUID exileId, String name, Ability source, Game game, List<UUID> appliedEffects) {
-        Zone fromZone = game.getState().getZone(objectId);
-        ZoneChangeEvent event = new ZoneChangeEvent(this, source, ownerId, fromZone, Zone.EXILED, appliedEffects);
-        ZoneChangeInfo.Exile zcInfo = new ZoneChangeInfo.Exile(event, exileId, name);
+        if (!game.replaceEvent(GameEvent.getEvent(EventType.EXILE_PERMANENT, objectId, source, controllerId))) {
+            Zone fromZone = game.getState().getZone(objectId);
+            ZoneChangeEvent event = new ZoneChangeEvent(this, source, ownerId, fromZone, Zone.EXILED, appliedEffects);
+            ZoneChangeInfo.Exile zcInfo = new ZoneChangeInfo.Exile(event, exileId, name);
 
-        boolean successfullyMoved = ZonesHandler.moveCard(zcInfo, game, source);
-        //20180810 - 701.3d
-        detachAllAttachments(game);
-        return successfullyMoved;
+            boolean successfullyMoved = ZonesHandler.moveCard(zcInfo, game, source);
+            //20180810 - 701.3d
+            detachAllAttachments(game);
```

Comment:
BTW that's `detachAllAttachments` smells bad without `successfullyMoved` check -- looks like it can broke `Bestow` ability (permanent can't be removed due protection, but attached aura does):

![shot_231028_140205](https://github.com/magefree/mage/assets/8344157/ca1cd57b-132a-470c-9916-8050e1b3885e)

Reply:
Would writing
```java
if (successfullyMoved) {
    detachAllAttachments(game);
}
```
Be sufficient?

Reply:
Yes, its ok. Add it to all detachAllAttachments